### PR TITLE
Added Key collection feature

### PR DESCRIPTION
**Pull request by vighnajeet**

### CODE REVIEW

**Comment by vighnajeet** [comment]
Hi Boris,

I had created the pull request couple of days back. Yesterday while
checking on Jijo's laptop sent the review request by mistake from there. I
am working on it individually.

Regards,
Vighnajeet Naik

File: src/test/resources/features/main/KeyCollection.feature [hunk 1]
```diff
@@ -0,0 +1,12 @@
+Feature: Player collects the key
+
+Scenario: Player moves on key to collect it
+    Given the level of game is:
+| XXXXXXXXXXX |
+| X         X |
```
Comment (borisv13) [inline src/test/resources/features/main/KeyCollection.feature]: Does line 6 need to exist?

File: src/main/java/engine/GameEngine.java [hunk 2]
```diff
@@ -94,7 +94,14 @@ private void movePlayer(int xDiff, int yDiff) {
 				getPlayerYCoordinate() + yDiff);
 		if (attempedLocation.equals(TileType.PASSABLE)) {
 			setPlayer(getPlayerXCoordinate() + xDiff, getPlayerYCoordinate() + yDiff);
+		} else if (attempedLocation.equals(TileType.KEY)) {
+			setPlayer(getPlayerXCoordinate() + xDiff, getPlayerYCoordinate() + yDiff);
+			tiles.remove(new Point(1, 2), TileType.KEY);
+			tiles.put(new Point(1, 2), TileType.PASSABLE);
+		} else if (attempedLocation.equals(TileType.DOOR) && getTileFromCoordinates(1, 2) != TileType.KEY) {
```
Comment (borisv13) [inline src/main/java/engine/GameEngine.java]: What is 1,2? Why is that special? Whatever your answer to that question is, how can I know that by reading the code?

File: src/test/resources/features/main/DoorPassing.feature [hunk 3]
```diff
@@ -0,0 +1,11 @@
+Feature: Player passes through the door
+
+Scenario: Player moves towards the door to pass it
+    Given the level of game with door is:
+| XXXXXXXXXXX |
+| X   D     X |
+| X   P     X |
+| XXXXXXXXXXX |
+    When the key is not loacted on location (2, 3)
```
Comment (borisv13) [inline src/test/resources/features/main/DoorPassing.feature]: typo: loacted
This When is not an action. It's setting up data, which is a Given.

File: src/test/resources/features/main/DoorPassing.feature [hunk 4]
```diff
@@ -0,0 +1,11 @@
+Feature: Player passes through the door
+
+Scenario: Player moves towards the door to pass it
+    Given the level of game with door is:
+| XXXXXXXXXXX |
+| X   D     X |
+| X   P     X |
+| XXXXXXXXXXX |
+    When the key is not loacted on location (2, 3)
+    Then the player is located on door location (5, 2)
```
Comment (borisv13) [inline src/test/resources/features/main/DoorPassing.feature]: Again, why 5,2? That's extremely unclear! *Make level smaller*

File: src/test/resources/features/main/KeyCollection.feature (hunk 1 above)
Comment (borisv13) [reply src/test/resources/features/main/KeyCollection.feature]: Does the level have to be 10 tiles across?

File: src/main/java/engine/GameEngine.java [hunk 5]
```diff
@@ -94,7 +94,14 @@ private void movePlayer(int xDiff, int yDiff) {
 				getPlayerYCoordinate() + yDiff);
 		if (attempedLocation.equals(TileType.PASSABLE)) {
 			setPlayer(getPlayerXCoordinate() + xDiff, getPlayerYCoordinate() + yDiff);
+		} else if (attempedLocation.equals(TileType.KEY)) {
+			setPlayer(getPlayerXCoordinate() + xDiff, getPlayerYCoordinate() + yDiff);
+			tiles.remove(new Point(getPlayerXCoordinate(), getPlayerYCoordinate()), TileType.KEY);
+			tiles.put(new Point(getPlayerXCoordinate(), getPlayerYCoordinate()), TileType.PASSABLE);
+		} else if (attempedLocation.equals(TileType.DOOR) && getTileFromCoordinates(1, 1) != TileType.KEY) {
```
Comment (borisv13) [inline src/main/java/engine/GameEngine.java]: Extract into method, readable code.
Why 1,1? You made it so the game will only work is the key starts at 1,1? Gross.

File: src/test/resources/features/main/DoorPassing.feature [hunk 6]
```diff
@@ -0,0 +1,11 @@
+Feature: Player passes through the door
+
+Scenario: Player moves towards the door to pass it
+    Given the level of game with door is:
+| XXXXXX |
+| X  DPX |
+| XXXXXX |
+
+    When the player moves towards the door
```
Comment (borisv13) [inline src/test/resources/features/main/DoorPassing.feature]: The player can open the door without a key? That's not what we discussed.
And there is no Given that the player had picked up the key already.

**Pull request closed without merging**